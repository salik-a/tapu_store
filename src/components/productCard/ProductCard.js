import React from 'react';
import { Text, View } from 'react-native';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import Ionicons from 'react-native-vector-icons/Ionicons';
import { Container, Title, Description, NumberText, RowContainer, Image, InnerContainer, RedText, Vector, BottomContainer } from "./ProductCardStyled"
const ProductCard = ({ title, description, image, star, distance }) => {
    return (
        <Container>
            <RowContainer>
                <Image source={{
                    uri: 'https://reactnative.dev/img/tiny_logo.png',
                }} />
                <InnerContainer>
                    <Title>Villa Bosphorus</Title>
                    <Description>Lorem Ä°psum Sit Dolor Met</Description>
                    <RowContainer>
                        <RowContainer>
                            <FontAwesome name="star-half-empty" color={"#ECD03F"} size={24} />
                            <NumberText>3.9</NumberText>
                        </RowContainer>
                        <RowContainer>
                            <Ionicons name="location" color={"#0DAFC0"} size={22} />
                            <NumberText>3.7 km</NumberText>
                        </RowContainer>
                    </RowContainer>
                </InnerContainer>
            </RowContainer>
            <BottomContainer>
                <RowContainer>
                    <Vector source={require('../../assets/Vector.png')} />
                    <RedText>SEPETE EKLE</RedText>
                </RowContainer>
            </BottomContainer>
        </Container>
    );
};

export default ProductCard;