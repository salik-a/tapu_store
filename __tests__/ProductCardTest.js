import React from 'react'
import { fireEvent, render, waitFor } from '@testing-library/react-native'
import ProductCard from '../src/components/productCard/ProductCard';
test('should match with snapshot', () => {

    const data = {
        "title": "Villa Bosphorus",
        "image": "https://lh5.googleusercontent.com/-NUVFE9PlpeM/TWu1toPnthI/AAAAAAAAAAc/Rzj59I3aZj4/s320/villa+foto%25C4%259Fraflar%25C4%25B1.jpg",
        "star": 4,
        "distance": 81,
        "detail": "If you are looking for a bright and comfortable villa you found a right place.",
        "id": "1",
        "price": 100
    }
    const idList = ["1"]
    const comp = render(<ProductCard data={data} idList={idList} />);
    expect(comp).toMatchSnapshot();
});

test('should trigger sepete ekle button onPress', () => {

    const data = {
        "title": "Villa Bosphorus",
        "image": "https://lh5.googleusercontent.com/-NUVFE9PlpeM/TWu1toPnthI/AAAAAAAAAAc/Rzj59I3aZj4/s320/villa+foto%25C4%259Fraflar%25C4%25B1.jpg",
        "star": 4,
        "distance": 81,
        "detail": "If you are looking for a bright and comfortable villa you found a right place.",
        "id": "1",
        "price": 100
    }
    const idList = ["1"]
    const mockFunction = jest.fn()
    const comp = render(<ProductCard data={data} idList={idList} onPress={mockFunction} />);
    const button = comp.getByTestId("sepete-ekle")
    fireEvent(button, "press")
    expect(mockFunction).toBeCalledTimes(1)
});

