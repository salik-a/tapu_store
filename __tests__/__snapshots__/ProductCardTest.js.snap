// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
    ]
  }
  testID="product-card"
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://lh5.googleusercontent.com/-NUVFE9PlpeM/TWu1toPnthI/AAAAAAAAAAc/Rzj59I3aZj4/s320/villa+foto%25C4%259Fraflar%25C4%25B1.jpg",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 20,
            "height": 110,
            "marginHorizontal": 25,
            "width": 110,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontFamily": "Nunito Sans",
              "fontSize": 20,
              "fontWeight": "500",
            },
          ]
        }
      >
        Villa Bosphorus
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "gray",
              "fontSize": 14,
              "marginVertical": 10,
              "width": 200,
            },
          ]
        }
      >
        If you are looking for a bright and comfortable villa you found a right place.
      </Text>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#ECD03F",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 16,
                  "fontWeight": "300",
                  "marginLeft": 5,
                  "marginRight": 15,
                },
              ]
            }
          >
            4
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#0DAFC0",
                  "fontSize": 22,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 16,
                  "fontWeight": "300",
                  "marginLeft": 5,
                  "marginRight": 15,
                },
              ]
            }
          >
            81
             km
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "justifyContent": "center",
        "marginTop": 15,
        "opacity": 1,
      }
    }
    testID="sepete-ekle"
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/Vector.png",
          }
        }
        style={
          Array [
            Object {
              "height": 25,
              "resizeMode": "contain",
              "width": 25,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#E82223",
              "fontSize": 18,
              "fontWeight": "700",
              "marginLeft": 10,
            },
          ]
        }
        testID="button-text"
      >
        SEPETTEN ÇIKAR
      </Text>
    </View>
  </View>
</View>
`;
